js_modules/dagit/packages/core/src/assets/groupByPartition.tsx

```diff
@@ -34,10 +36,32 @@ export const groupByPartition = (
   return partitionKeys
     .filter((key) => key !== NO_PARTITION_KEY)
     .map((key) => {
-      const [latest, ...predecessors] = [...(grouped[key] || [])].sort(sortByEventTimestamp);
+      const sorted = [...(grouped[key] || [])].sort(sortByEventTimestamp);
+      const latestMaterialization = sorted.find((a) => a.__typename === 'MaterializationEvent');
+      let latest = latestMaterialization || sorted[0] || null;
+
+      if (latest && latest.__typename === 'MaterializationEvent') {
+        const observationsAboutLatest = sorted.filter(
+          (e) =>
+            e.__typename === 'ObservationEvent' && Number(e.timestamp) > Number(latest.timestamp),
+        );
+
+        latest = {...latest, metadataEntries: [...latest.metadataEntries]};
+        for (const observation of observationsAboutLatest) {
+          latest.metadataEntries.push(
+            ...observation.metadataEntries.map((e) => ({
+              ...e,
+              description: `Observed by ${observation.stepKey}${e.description?.length ? ': ' : ''}${
```

Comment:
Nit: Would it be possible to add the observation icon here e.g. "Observation from my_asset_2"? The "observed by" line is pretty subtle and its something I don't want users to miss, and it might help to add the icon for visual hinting.

Reply:
Sounds good! I updated this to look like this and include the run and observation time info:
![image](https://user-images.githubusercontent.com/1037212/152100869-72ffdb97-7f2a-4042-a5b4-165b8db78ca6.png)